### PR TITLE
[CDAP-6662] Logistic Regression Trainer implementation

### PR DESCRIPTION
**Pull request by romy-khetan**

### CODE REVIEW

File: spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/LogisticRegressionTrainer.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    @Nullable
+    @Description("The number of features to use in training the model. It must be of type integer and equal to the" +
+      " number of features used in LogisticRegressionClassifier. The default value if none is provided will be" +
+      " 100.")
+    private final Integer numFeatures;
+
+    @Description("The number of classes to use in training the model. It must be of type integer. " +
+      "The default value if none is provided will be 2.")
+    private final Integer numClasses;
```
Comment (knazir) [inline spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/LogisticRegressionTrainer.java:86]: Is this missing an @Nullable tag? "If none is provided..."

File: spark-plugins/docs/LogisticRegressionClassifier-sparkcompute.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# LogisticRegressionClassifier
+
+
+Description
+-----------
+Loads a Logistic Regression model from a file of a FileSet dataset and uses it to classify records.
```
Comment (abhinavbansal246) [inline spark-plugins/docs/LogisticRegressionClassifier-sparkcompute.md:6]: why is it limited to File Set? Can we extend it to get input from other sources too ?

File: spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/LogisticRegressionClassifier.java [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+ * SparkCompute that uses a trained Logistic Regression model to classify and tag input records.
+ */
+@Plugin(type = SparkCompute.PLUGIN_TYPE)
+@Name(LogisticRegressionClassifier.PLUGIN_NAME)
+@Description("Uses a trained Logistic Regression model to classify records.")
+public class LogisticRegressionClassifier extends SparkCompute<StructuredRecord, StructuredRecord> {
+
+  public static final String PLUGIN_NAME = "LogisticRegressionClassifier";
+
+  private Config config;
```
Comment (abhinavbansal246) [inline spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/LogisticRegressionClassifier.java]: make final

File: spark-plugins/docs/LogisticRegressionClassifier-sparkcompute.md (hunk 2 above)
Comment (anwar6953) [inline spark-plugins/docs/LogisticRegressionClassifier-sparkcompute.md:6]: it has to be file-based.
Maybe PartitionedFileSet would also work, but I think its fine to support just FileSet for now.

File: spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/LogisticRegressionTrainer.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    if (inputSchema != null) {
+      Schema.Type predictionFieldType = inputSchema.getField(config.predictionField).getSchema().getType();
+      Preconditions.checkArgument(predictionFieldType == Schema.Type.INT,
+                                  "Prediction field must be of type Int, but was %s.", predictionFieldType);
+    }
+  }
+
+  @Override
+  public void prepareRun(SparkPluginContext context) throws Exception {
```
Comment (anwar6953) [inline spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/LogisticRegressionTrainer.java:112]: do we need to override this method, then?

Comment (knazir) [inline spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/LogisticRegressionTrainer.java]: `SparkCompute` does not implement `prepareRun` is this not giving you compiler warnings?

Comment (knazir) [inline spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/LogisticRegressionTrainer.java:112]: `SparkSink` does not implement `prepareRun` and it's abstract in `BatchConfigurable` so it has to be implemented here.

Comment (romy-khetan) [inline spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/LogisticRegressionTrainer.java:112]: Logistic Regression Trainer does not need to set any job configs in prepareRun, that's why left this function blank.

**Comment by abhinavbansal246** [comment]
Closing PR against release, please open against develop

**Pull request closed without merging**